src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -1844,6 +1845,11 @@ size_t G1CollectedHeap::used() const {
   return result;
 }
 
+size_t G1CollectedHeap::live() const {
+  size_t size = Atomic::load(&_live_size);
+  return size > 0 ? size : used();
```

Comment:
note that `used()` is susceptible to fluttering due to memory ordering problems: since its result consists of multiple reads, you can get readings from very different situations.
It is recommended to use `used_unlocked()` instead, which does not take allocation regions and archive regions into account, but at least it is not susceptible to jumping around when re-reading it in quick succession.